firebase/src/core/utils/validate.ts

```diff
@@ -11,7 +11,7 @@ export const defaultMaxStringLength = 255;
  * @param allowUnknown Allows object to contain unknown keys which are ignored.
  * @param convert Attempts to cast values to the required types.
  */
-export function validate(schema: Joi.Schema, data: unknown, allowUnknown = true, convert = false): void {
+export function validate(schema: Joi.Schema, data: unknown, allowUnknown = false, convert = false): void {
```

Comment:
Here I strongly suggest to be less permissive by default. I had some false-positives in the tests when turning this to `false` that made me think that we should be explicit when we want to allow unknown properties, and not the opposite.